shared/utils/relabel-ids.py

```diff
@@ -223,22 +223,21 @@ def _ensure_by_xccdf_referenced_oval_def_is_defined_in_oval_file(
                 # The OVAL check was found, we can continue
                 continue
 
-            if len(self.checks_related_to_us) == 0:
-                # Skip XCCDF rules not referencing OVAL checks
-                continue
-            check = self.checks_related_to_us.pop()
+            for check in rule.findall(".//{%s}check" % (xccdf_ns)):
+                if check.get("system") != oval_cs:
+                    continue
 
-            if get_check_content_ref_if_exists_and_not_remote(check) is not None:
-                continue
+                if get_check_content_ref_if_exists_and_not_remote(check) is None:
```

Comment:
I am not a big fan of the name of this function. so it returns None if it doesn't exist, it returns None if it's remote, does the logic around it work then?